### PR TITLE
Add Prettier

### PR DESCRIPTION
**Pull request by southpolesteve**
- Adds prettier with default behavior
- Adds script `npm run format` for use in dev
- Adds script `npm run check-format` for use in CI

Prettier has two defaults which I'm not opposed to changing: 
- 80 character line length
- Double quotes for strings

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,6 +20,8 @@
   },
   "scripts": {
     "clean": "rimraf lib",
+    "format": "prettier --write \"src/**/*.ts\"",
+    "check-format": "prettier --list-different \"src/**/*.ts\"",
```
Comment (christopheranderson) [inline package.json]: I think we should have the build fail if there is formatting issues (similar to lint failures). Could this get added to the build command?

Comment (southpolesteve) [reply package.json]: Yep. I'll add

**Comment by southpolesteve** [comment]
There is no config right now. We can put it in `.prettierrc` or in `package.json#prettier`

**Comment by christopheranderson** [comment]
Let's put the config in now so its easier to configure later.

**Comment by christopheranderson** [comment]
Dealers choice on a separate file or package.json approach. I like both.

**Pull request merged**